### PR TITLE
Add support for using a Llama.cpp binary and model from TurnkeyML

### PR DESCRIPTION
**Pull request by gabeweisz**
This PR adds support for using running a Llama.cpp binary and model from TurnkeyML.
It has been tested using MMLU for evaluation.

This PR also gives MMLU the ability to limit the number of inputs evaluated when running tests.

The Llama.cpp invocation and output scraping code is based on code created by @ramkrishna2910

### CODE REVIEW

**Comment by gabeweisz** [comment]
Per @ramkrishna2910's request, will add llama.cpp compile instructions

**Comment by jeremyfowers** [comment]
Note: all PRs will require a fix like this to pass CI: https://github.com/onnx/turnkeyml/pull/238

File: src/turnkeyml/llm/docs/llamacpp.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# LLAMA.CPP
+
+Run transformer models using a Llama.cpp binary and checkpoint. This model can then be used with chatting or benchmarks such as MMLU.
+
+## Prerequisites
+
+This flow has been verified with a generic Llama.cpp model.
+
+### Set up Environment (Assumes TurnkeyML is already installed)
```
Comment (ramkrishna2910) [inline src/turnkeyml/llm/docs/llamacpp.md:9]: Please mention that these instructions are only for Linux/ WSL or add instructions for Windows with cmake

Comment (ramkrishna2910) [inline src/turnkeyml/llm/docs/llamacpp.md:9]: Please mention that pre-requisite is `pip install -e .[llm]`

File: src/turnkeyml/llm/docs/llamacpp.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Build or obtain the Llama.cpp model and desired checkpoint.
+For example (see the [llama.cpp](https://github.com/ggerganov/llama.cpp/blob/master/docs/build.md
+) source for more details):
```
Comment (ramkrishna2910) [inline src/turnkeyml/llm/docs/llamacpp.md:13]: On Windows, I needed to be on the Administrator prompt for this to work. It is worth noting here in the Windows instructions section.

File: src/turnkeyml/llm/docs/llamacpp.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+| executable   | Path to the Llama.cpp-generated application binary                          | None    |
+| model-binary | Model checkpoint (do not use if --input is passed to lemonade)              | None    |
+| threads      | Number of threads to use for computation                                    | 1       |
+| context-size | Maximum context length                                                      | 512     |
+| temp         | Temperature to use for inference (leave out to use the application default) | None    |
+
+### Example (assuming Llama.cpp built and a checkpoint loaded as above)
+
+```bash
+lemonade --input ~/llama.cpp/models/dolphin-llama2-7b.Q5_K_M.gguf load-llama-cpp --executable ~/llama.cpp/llama-cli accuracy-mmlu --ntrain 5
```
Comment (ramkrishna2910) [inline src/turnkeyml/llm/docs/llamacpp.md:37]: `llama-cli` is found inside `llama.cpp\build\bin\Release\`. Please edit command to reflect the path.

File: src/turnkeyml/llm/docs/llamacpp.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+This flow has been verified with a generic Llama.cpp model.
+
+### Set up Environment (Assumes TurnkeyML is already installed)
+
+Build or obtain the Llama.cpp model and desired checkpoint.
+For example (see the [llama.cpp](https://github.com/ggerganov/llama.cpp/blob/master/docs/build.md
+) source for more details):
+1. cd ~
+1. git clone git clone https://github.com/ggerganov/llama.cpp
```
[the remark this answers is not in the document]
Comment (gabeweisz) [reply src/turnkeyml/llm/docs/llamacpp.md:15]: Done

File: src/turnkeyml/llm/docs/llamacpp.md (hunk 1 above)
Comment (gabeweisz) [reply src/turnkeyml/llm/docs/llamacpp.md:9]: Done

Comment (gabeweisz) [reply src/turnkeyml/llm/docs/llamacpp.md:14]: Done

File: src/turnkeyml/llm/docs/llamacpp.md (hunk 2 above)
Comment (gabeweisz) [reply src/turnkeyml/llm/docs/llamacpp.md:18]: Done

File: src/turnkeyml/llm/docs/llamacpp.md (hunk 3 above)
Comment (gabeweisz) [reply src/turnkeyml/llm/docs/llamacpp.md:42]: For me, on Linux, the binary was just in llama.cpp, but I will add a comment about this.

**Pull request merged**